### PR TITLE
0.9.X Fixes

### PR DESCRIPTION
**Pull request by ianjennings**

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -60,9 +60,19 @@ module.exports = (config = {}) => {
...
 
-        if(payload && payload.sender && payload.chat && payload.chat.muter.isMuted(payload.sender)) {
+        let isOwnEvent = false;
 
-            payload.chat.trigger('$.muter.eventRejected', payload);
+        if(payload && payload.event && payload.event.indexOf('$.muter') > -1) {
+            isOwnEvent = true;
+        }
+
+        if(!isOwnEvent && payload && payload.sender && payload.chat && payload.chat.muter && payload.chat.muter.isMuted(payload.sender)) {
```
Comment (crgbaumgart) [inline src/plugin.js:69]: Thats a mighty big ```if()``` conditional you got there.

File: src/plugin.js [hunk 2]
```diff
@@ -76,7 +86,8 @@ module.exports = (config = {}) => {
         namespace: 'muter',
         extends: {
             Chat: extension
-        },
+        }
```
Comment (crgbaumgart) [inline src/plugin.js:89]: Why'd the comma get its own line? Tiny syntax nit-pick

File: test.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 const assert = require('chai').assert;
 const mute = require('./src/plugin.js');
 
-const ChatEngine = require('../chat-engine/src/index.js');
+const ChatEngine = require('./node_modules/chat-engine/src/index.js');
```
Comment (crgbaumgart) [inline test.js:5]: 👍

File: test.js [hunk 4]
```diff
@@ -12,11 +12,8 @@ describe('config', function() {
     it('should be configured', function() {
 
         CE = ChatEngine.create({
-            publishKey: 'pub-c-c6303bb2-8bf8-4417-aac7-e83b52237ea6',
-            subscribeKey: 'sub-c-67db0e7a-50be-11e7-bf50-02ee2ddab7fe',
-        }, {
-            endpoint: 'http://localhost:3000/insecure',
-            globalChannel: 'test-channel'
+            publishKey: 'pub-c-01491c54-379f-4d4a-b20b-9a03c24447c7',
```
Comment (crgbaumgart) [inline test.js:15]: I'm not gonna block you cause these keys are already in GH, but we should try to remove these or encrypt the values.

**Review by crgbaumgart — APPROVED** [review]
Left a few comments, but nothing that requires addressing now.

File: src/plugin.js (hunk 2 above)
Comment (ianjennings) [reply src/plugin.js:89]: Was commenting out lines

File: src/plugin.js (hunk 1 above)
Comment (ianjennings) [reply src/plugin.js:69]: :*(

I know, it should be multiple variable assignment then a check for the result of those.

**Pull request merged**